casper/src/main/resources/PoS.rho

```diff
@@ -1,180 +1,165 @@
-                          } else return!((true, bondAmount))
-                        }
-                      }
-                    }
-                  }
-
-                  | contract updateBonds (@bondAmount, return) = {
-                    new parametersCh in {
-                      getParameters!(*parametersCh)
-                      | for(@bonds <- bondsCh; _, _, @userPk, _ <- parametersCh) {
-                        bondsCh!(
-                          bonds.set(
-                            userPk,
-                            [bondAmount, bonds.size() + 1]
-                          )
-                        ) | return!((true, "Bond successful!"))
-                      }
+            } |
+
+            contract PoS (@"bond", @amount, returnCh) = {
+              new userCh, depositCh in {
+                getUser!(*userCh) |
+
+                for(@bonds <- bondsCh;
+                    @userPk <- userCh) {
+
+                  if (bonds.contains(userPk)) {
+                    bondsCh!(bonds) |
+                    returnCh!((false, "Public key is already bonded."))
+                  } else if (amount < minimumBond) {
+                    bondsCh!(bonds) |
+                    returnCh!((false, "Bond less than minimum!"))
+                  } else if (amount > maximumBond) {
+                    bondsCh!(bonds) |
+                    returnCh!((false, "Bond greater than maximum!"))
+                  } else {
+                    deposit!(userPk, amount, *depositCh) |
+                    for(@depositResult <- depositCh) {
+                      bondsCh!(
+                        bonds.set(
+                          userPk,
+                          [amount, bonds.size() + 1]
+                        )
+                      ) |
+                      returnCh!(depositResult)
                     }
                   }
                 }
               }
+            } |
 
-              //Assuming that user public key is valid, a vault should always be returned.
-              | contract deposit (@amount, return) = {
-                new parametersCh,
-                  vaultCh,
-                  revAddressCh,
-                  authKeyCh in {
-                  getParameters!(*parametersCh)
-                  | for (_, _, @userPk, _ <- parametersCh) {
-                    revAddressOps!("fromPublicKey", userPk, *revAddressCh)
-                    | for (@revAddress <- revAddressCh) {
-                      @RevVault!("findOrCreate", revAddress, *vaultCh)
-                      | for (@(true, vault) <- vaultCh) {
-                        @RevVault!("deployerAuthKey", *authKeyCh)
-                        | for (userAuthKey <- authKeyCh){
-                          @vault!("transfer", posRevAddress, amount, *userAuthKey, *return)
-                        }
-                      }
-                    }
-                  }
-                }
-              }
+            contract PoS (@"pay", @amount, returnCh) = {
```

Comment:
What does this PoS.pay do? Is this supposed to be the basic template for rewards that will be expanded later?

Reply:
Oh nvm...; this will conflict with

https://github.com/rchain/rchain/blob/f4980d38f183d80b02ecd9a3fe59ca64d2552f93/casper/src/main/resources/PoS.rho#L155

Reply:
It looks like `lastPayment` in the `MakePoS` contract is only used here [https://github.com/rchain/rchain/blob/f4980d38f183d80b02ecd9a3fe59ca64d2552f93/casper/src/test/scala/coop/rchain/casper/MultiParentCasperDeploySpec.scala#L188](url). Once we have vaults and gas linked, we won't need this test anymore, so it's ok to remove.

Reply:
I don't understand this thread. Please explain.